standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -803,6 +765,114 @@ private void open() throws MetaException {
     snapshotActiveConf();
   }
 
+  private String[] getHttpUrlAndUser(URI store) throws MetaException {
+    boolean useSSL = MetastoreConf.getBoolVar(conf, ConfVars.USE_SSL);
+    String path = MetaStoreUtils.getHttpPath(MetastoreConf.getVar(conf, ConfVars.THRIFT_HTTP_PATH));
+    String httpUrl = (useSSL ? "https://" : "http://") + store.getHost() + ":" + store.getPort() + path;
+    String httpUser = MetastoreConf.getVar(conf, ConfVars.METASTORE_CLIENT_PLAIN_USERNAME);
```

Comment:
we probably want to make sure that this config can only be changed by an admin

Reply:
Makes sense. How do we ensure that for other configs? Would take it up in a follow up patch.